community/dbms/src/main/java/org/neo4j/commandline/dbms/MemoryRecommendationsCommand.java

```diff
@@ -0,0 +1,385 @@
+import org.neo4j.commandline.admin.OutsideWorld;
+import org.neo4j.commandline.arguments.Arguments;
+import org.neo4j.commandline.arguments.OptionalBooleanArg;
+import org.neo4j.commandline.arguments.OptionalNamedArg;
+import org.neo4j.helpers.collection.MapUtil;
+import org.neo4j.kernel.configuration.Config;
+import org.neo4j.kernel.impl.util.OsBeanUtil;
+
+import static java.lang.String.format;
+import static org.neo4j.configuration.ExternalSettings.initialHeapSize;
+import static org.neo4j.configuration.ExternalSettings.maxHeapSize;
+import static org.neo4j.graphdb.factory.GraphDatabaseSettings.pagecache_memory;
+import static org.neo4j.io.ByteUnit.ONE_GIBI_BYTE;
+import static org.neo4j.io.ByteUnit.ONE_KIBI_BYTE;
+import static org.neo4j.io.ByteUnit.ONE_MEBI_BYTE;
+import static org.neo4j.io.ByteUnit.gibiBytes;
+import static org.neo4j.io.ByteUnit.mebiBytes;
+import static org.neo4j.io.ByteUnit.tebiBytes;
+import static org.neo4j.kernel.configuration.Settings.BYTES;
+import static org.neo4j.kernel.configuration.Settings.buildSetting;
+
+public class MemoryRecommendationsCommand implements AdminCommand
+{
+    // Fields: {System Memory in GiBs; OS memory reserve in GiBs; JVM Heap memory in GiBs}.
+    // And the page cache gets what's left, though always at least 100 MiB.
+    // Heap never goes beyond 32 GiBs, which is the upper limit for compressed oops..
+    private static final Bracket[] datapoints = {
+            new Bracket( 0.01, 0.005, 0.005 ),
+            new Bracket( 1.0, 0.5, 0.5 ),
+            new Bracket( 2.0, 1, 0.5 ),
+            new Bracket( 4.0, 1.5, 2 ),
+            new Bracket( 6.0, 2, 3 ),
+            new Bracket( 8.0, 2.5, 3.5 ),
+            new Bracket( 10.0, 3, 4 ),
+            new Bracket( 12.0, 3.5, 4.5 ),
+            new Bracket( 16.0, 4, 5 ),
+            new Bracket( 24.0, 4.5, 8 ),
+            new Bracket( 32.0, 5, 12 ),
+            new Bracket( 64.0, 6, 24 ),
+            new Bracket( 128.0, 8, 32 ),
+            new Bracket( 256.0, 12, 32 ),
+            new Bracket( 512.0, 16, 32 ),
+            new Bracket( 1024.0, 30, 32 ),
+    };
+    private static final String ARG_MEMORY = "memory";
+    private static final String ARG_PRINT_CONF = "print-conf";
+    private static final String ARG_UPDATE_CONF = "update-conf";
+    private static final String ARG_OVERWRITE = "overwrite";
+    private static final String ARG_CONF_FILE = "conf-file";
+    private final OutsideWorld outsideWorld;
+    private final Path configDir;
+
+    static long recommendOsMemory( long totalMemoryBytes )
+    {
+        Brackets brackets = findMemoryBrackets( totalMemoryBytes );
+        return brackets.recommend( Bracket::osMemory );
+    }
+
+    static long recommendHeapMemory( long totalMemoryBytes )
+    {
+        Brackets brackets = findMemoryBrackets( totalMemoryBytes );
+        return brackets.recommend( Bracket::heapMemory );
+    }
+
+    static long recommendPageCacheMemory( long totalMemoryBytes )
```

Comment:
We have a heuristic currently that tries to determine correct page cache size in case if that was not configured explicitly. We need to merge those two things and have only one "recommender"

Reply:
As I can see one part of the recommendation is gone, but heuristic that calculates size still there. Shouldn't our recommended value and heuristic evaluated be the same and use the same code for evaluation?